### ISSUE TITLE
Infinite loop when building and index that has been empty

### ISSUE THREAD

**Issue by lerouxrgd** [yahoojapan/NGT#68]
Hello, during one of my tests I tried the following scenario:

- Create a new empty index
- Insert one vector
- Build the index
- Remove the vector // index size is 0 here
- (Build or not the index, it makes no difference)
- Insert a vector (same as before or new one, it make no difference)
- Build the index // this will infinite loop

With Rust code it looks like this:

```rust
// Get a temporary directory to store the index
let dir = tempfile::tempdir()?;

// Create an index for vectors of dimension 3
let prop = ngt::Properties::new(3)?;
let mut index = ngt::Index::create(dir.path(), prop)?;

// Insert a vector
let vec1 = vec![1.0, 2.0, 3.0];
let id1 = index.insert(vec1.clone())?;

// Build index
index.build(2)?;

// Remove the vector, index size is now 0
index.remove(id1)?;

// Add one vector, index size should be 1 after build
// (Here we use vec1, but using a different vector triggers same issue)
index.insert(vec1)?;

// This will be an infinite loop
index.build(2)?;
```

Should we explicitly delete an index from disk when its size reaches 0 and create a new one instead ?

**Comment by masajiro** [comment]
I released NGT v1.11.6 which fixed this bug.

**Comment by lerouxrgd** [comment]
Thank you for fixing the issue !
